lib/ws_servers/api/algos/algo_worker.js

```diff
@@ -162,11 +150,20 @@ class AlgoWorker {
   async cancelOrder (gid) {
     const { host } = this
 
-    if (!host.getAOInstance(gid)) {
+    const instance = host.getAOInstance(gid)
+    if (!instance) {
       throw new Error('Requested algo order not running, cannot stop')
     }
 
+    const serialized = host.getSerializedAO(instance)
+    serialized.active = false
+    this._updateAlgo(serialized)
+
     await host.stopAO(gid)
```

Comment:
or may be update after it's stopped ?
since `serialized` can be returned from stopAO as well ?

Reply:
with this code i want to ensure that the entry in the database is always set to inactive... imagine `stopAlgo` hanging or crashing, then on restart the algo would be still active...

Reply:
hmm.. thought so.. may be it's not required in `stopAlgo` then?

Reply:
```js
    await withAOUpdate(this, gid, (instance = {}) => {
      const { state = {} } = instance
      state.active = false

      return false
    })
 ```

sets the state to false, but because false is returned, no update / former db write is emitted

Reply:
see here:

https://github.com/bitfinexcom/bfx-hf-algo/blob/32a226b152ee175c7400258dc4158a3cf7fde93f/lib/host/with_ao_update.js#L24-L27